js/TableDetailComponent.js

```diff
@@ -18,22 +18,39 @@ class TableDetailComponent {
         }
 
         let $panelHtml = $(
-            '<div class="table-detail t-table-detail" data-table-detail-panel>' +
+            '<div class="table-detail t-table-detail" data-table-detail-panel role="dialog" aria-modal="true" aria-hidden="true">' +
             '   <div class="table-detail__header">' +
-            '       <button type="button" class="close table-detail__header-close" data-table-detail-close-panel aria-hidden="true">&times;</button>' +
+            '       <button type="button" class="close table-detail__header-close" data-table-detail-close-panel aria-label="Close" tabindex="-1"><span aria-hidden="true">&times;</span></button>' +
             '       <h1 class="table-detail__title" data-table-detail-panel-title>Detail</h1>' +
             '   </div>' +
             '   <div class="table-detail__body" data-table-detail-panel-body></div>' +
             '</div>'
-        );
+            ),
+            $panelAlreadyExists = this.$html.find('[data-table-detail-panel]'),
+            $main = this.$html.find('main'),
+            $roleMain = this.$html.find('[role="main"]'),
+            $elementToAppendTo,
+            $triggeringElement;
 
         this.$table = this.$html.find('[data-table-detail-table]');
 
-        // Add backdrop and detail panel if UI contains a table detail pattern
-        if (this.$table.length) {
-            this.$html.find('body')
-                .append('<div class="table-detail-backdrop"></div>')
-                .append($panelHtml);
+        // Check panel hasn't already been added, in case of a datatable re-initting this component on paginated page draw
+        if ($panelAlreadyExists.length === 0) {
```

Comment:
shorter with `if (!$panelAlreadyExists.length) {` ?

Reply:
Personally I prefer the explicitness of `=== 0`. I can change if we really feel this is an issue.